git-sync/atomic.md

```diff
@@ -0,0 +1,73 @@
+# Atomic initial clone
+
+To encourage sidecar functionality, git-sync now provides the option to make your initial clone atomic. Many use cases would require that the target repo directory be completely populated before another container begins accessing the content. 
+
+For example, a webserver container might run alongside the git-sync sidecar and serve content from the Git repo. The webserver would not want to begin serving the content in its target directory until the data is fully pulled into the volume by git-sync.  The process involves cloning and pulling initially to a temporary directory, setting a symbolic link, and then renaming the symbolic link to the desired target name. We can ensure that the target directory name will only appear to the webserver in the volume filesystem once the pull is complete.
+
+## Usage
+
+In the configuration of the git-sync container, set the optional GIT_SYNC_ATOMIC environment variable to true. The GIT_SYNC_SUBDIR environment variable must also be set to a non-empty string (otherwise an error will result). This subdirectory will be created within the shared volume's root directory, specified by GIT_SYNC_DEST (which defaults to "/git" if unspecified). If specified, GIT_SYNC_DEST should match the mountPath for git-sync's mounted emptyDir volume.
+
+The subdirectory is necessary because the volume root directory path cannot be replaced, but we can do so for a symbolic link to a subdirectory. Any other containers using the volume's content should look for it in GIT_SYNC_SUBDIR within their own mountPath. For example, a webserver that uses the shared volume configured below might mount the volume at "/shared" in its own filesystem, and would look for the repo content at "/shared/data".
+```
+{
+   ...
+   containers: [
+   {
+       name: "git-sync",
+       ...
+       env: [{
+               name: "GIT_SYNC_REPO",
+               value: "https://github.com/kubernetes/kubernetes",
+           },
+           {
+               name: "GIT_SYNC_SUBDIR",
+               value: "data",
+           },
+           {
+               name: "GIT_SYNC_ATOMIC",
+               value: "true",
+           },
+       ],
+       volumeMounts: [
+           {
+               name: "git-data",
+               mountPath: "/git",
+           },
+       ],
+   }],
+   ...
+   volumes: [
+       {
+           name: "git-data",
+           emptyDir: {},
+       },
+   ],
+   ...
+}
+```
+
+## Code Implementation
```

Comment:
Looks like the paragraphs and the pseudocode are explaining the same thing? If so, I'd keep the paragraphs high-level (e.g. don't refer to local variable names) and let the pseudocode cover the nitty-gritty.